receiver/snmpreceiver/scraper.go

```diff
@@ -39,13 +66,364 @@ func newScraper(logger *zap.Logger, cfg *Config, settings component.ReceiverCrea
+		attributeKey := attribute.Name
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+		dataPointAttributes[attributeKey] = attribute.Value
+	}
+
+	return dataPointAttributes
+}
+
+// getDataPointAttributes gets attributes for this metric's datapoint based on the previously
+// gathered attributes.
+// Keys will be determined from the related attribute config and values will come a few
+// different places.
+// Enum attribute value - comes from the metric config's attribute data
+// Indexed prefix attribute value - comes from the current SNMP data's index and the attribute
+// config's prefix value
+// Indexed OID attribute value - comes from the previously collected indexed attribute data
+// using the current index and attribute config to access the correct value
+func getIndexedDataPointAttributes(
+	configHelper *configHelper,
+	parentOID string,
+	indexString string,
+	indexedAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	datapointAttributes := map[string]string{}
+
+	for _, attribute := range configHelper.getMetricConfigAttributes(parentOID) {
+		attributeKey := attribute.Name
+		attributeValue := attribute.Value
+		if prefix := configHelper.getAttributeConfigIndexedValuePrefix(attributeKey); prefix != "" {
+			attributeValue = prefix + indexString
+		}
+		if oid := configHelper.getAttributeConfigOID(attributeKey); oid != "" {
+			indexedAttrKey := indexedAttributeKey{
+				parentOID: oid,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedAttributeValues[indexedAttrKey]
+		}
+		if attributeValue == "" {
+			return nil, errors.New(errMsgAttributeEmptyValue)
+		}
+		if value := configHelper.getAttributeConfigValue(attributeKey); value != "" {
+			attributeKey = value
+		}
+		datapointAttributes[attributeKey] = attributeValue
+	}
+
+	return datapointAttributes, nil
+}
+
+// getResourceAttributes creates a map of key/values for all related resource attributes. Keys
+// will come directly from the metric config's resource attribute values. Values will come
+// from the related attribute config's prefix value plus the index OR the previously collected
+// resource attribute indexed data.
+func getResourceAttributes(
+	configHelper *configHelper,
+	parentOID string,
+	indexString string,
+	indexedResourceAttributeValues map[indexedAttributeKey]string,
+) (map[string]string, error) {
+	resourceAttributes := map[string]string{}
+
+	for _, attributeName := range configHelper.getResourceAttributeNames(parentOID) {
+		var attributeValue string
+		if prefix := configHelper.getResourceAttributeConfigIndexedValuePrefix(attributeName); prefix != "" {
+			attributeValue = prefix + indexString
+		}
+		if oid := configHelper.getResourceAttributeConfigOID(attributeName); oid != "" {
+			indexedAttrKey := indexedAttributeKey{
+				parentOID: oid,
+				oidIndex:  indexString,
+			}
+			attributeValue = indexedResourceAttributeValues[indexedAttrKey]
+		}
```

Comment:
Is it possible for both of these conditions to be true? If so, is it intended that the second will overwrite the first, or should it append to the value?

Reply:
Ok, please use a control struct that enforces mutual exclusion on the conditions.